### ISSUE TITLE
"generates" does not handle absolute paths.

### ISSUE THREAD

**Issue by smyrman** [go-task/task#47]
It may seem like that the files listed in "generates" (and maybe the ones in "sources"; haven't checked) are prefixed with the task dir, even when an absolute path is given.

Below is a principal example for reproducing the problem:

Taskvars.yml:
```yaml
ROOT: $pwd
```

Taskfile.yml:
```yaml
default:
  dir: sub
  cmds:
    - cat file.txt > '{{.ROOT}}/file2.txt'
  sources:
    - file.txt
  generates:
    - "{{.ROOT}}/file2.txt"
```

sub/file.txt
```txt
hello world
```

If i run this task twice, it would not be found up to date:

``` sh
$ go get -u github.com/go-task/task/cmd/task   # Ensure task is up-to-date
$ task -v
task: No argument given, trying default task
task: dynamic variable: "$pwd", result: "/Users/smyrman/.local/src/github.com/searis/play"
cat file.txt > '/Users/smyrman/.local/src/github.com/searis/play/file2.txt'
$ task -v
task: No argument given, trying default task
task: dynamic variable: "$pwd", result: "/Users/smyrman/.local/src/github.com/searis/play"
cat file.txt > '/Users/smyrman/.local/src/github.com/searis/play/file2.txt'
```

If I _change_ the generates section in the Taskfile.yml to be a relative path, it works:
```yaml
  generates:
    - "../file2.txt"
```
Relaunch task:
```sh
$ task -v        
task: No argument given, trying default task
task: dynamic variable: "$pwd", result: "/Users/smyrman/.local/src/github.com/searis/play"
task: Task "default" is up to date
```

As for the use-case of absolute paths in generates, it matters when you want to use the same (configurable) output directory for several build targets that are built from various sub-directories.

**Comment by smyrman** [comment]
I will do a PR

**Comment by andreynering** [comment]
This fix is as simple as changing:

```go
p = filepath.Join(dir, p)
```

to

```go
if !filepath.IsAbs(p) {
        p = filepath.Join(dir, p)
}
```

[Here][here1] and [here][here2].

[here1]: https://github.com/go-task/task/blob/master/file.go#L26
[here2]: https://github.com/go-task/task/blob/master/file.go#L37

**Comment by smyrman** [comment]
> This fix is as simple as changing (...)

I found that pretty quick, but I used for ever to figure out why the new tests didn't work:-)

**Comment by smyrman** [comment]
All fixed now in PR #48.
